src/ucp/proto/proto_common.c

```diff
@@ -147,17 +147,35 @@ size_t ucp_proto_common_get_iface_attr_field(const uct_iface_attr_t *iface_attr,
     return *(const size_t*)UCS_PTR_BYTE_OFFSET(iface_attr, field_offset);
 }
 
+static int
+ucp_proto_common_check_lane_type(const ucp_ep_config_key_lane_t *lane,
+                                 ucp_lane_type_t lane_type)
+{
+    return lane->lane_types & UCS_BIT(lane_type);
+}
+
 static void
 ucp_proto_common_get_frag_size(const ucp_proto_common_init_params_t *params,
                                const uct_iface_attr_t *iface_attr,
-                               size_t *min_frag_p, size_t *max_frag_p)
+                               ucp_lane_index_t lane_idx, size_t *min_frag_p,
+                               size_t *max_frag_p)
 {
+    const ucp_ep_config_key_lane_t *lane;
+
     *min_frag_p = ucp_proto_common_get_iface_attr_field(iface_attr,
                                                         params->min_frag_offs,
                                                         0);
     *max_frag_p = ucp_proto_common_get_iface_attr_field(iface_attr,
                                                         params->max_frag_offs,
                                                         SIZE_MAX);
+
+    ucs_assert(lane_idx < UCP_MAX_LANES);
```

Comment:
1. pls add comment that we do it to not send more than what the remote side supports
2. i think we should decide about using seg_size according to the protocol logic of using this lane, and not according to the lane type. for example, pass uct opcode to ucp_proto_common_get_frag_size, deduce it from "params", or pass boolean parameter.